hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/safemode/SCMSafeModeManager.java

```diff
@@ -225,6 +245,63 @@ public double getCurrentContainerThreshold() {
         .getCurrentContainerThreshold();
   }
 
+  private synchronized void startSafeModePeriodicLogger() {
+    if (!getInSafeMode() || safeModeLogExecutor != null) {
+      return;
+    }
+    safeModeLogExecutor = Executors.newScheduledThreadPool(1,
+        new ThreadFactoryBuilder()
+            .setNameFormat(scmContext.threadNamePrefix() + "SCM-SafeMode-Log-%d")
+            .setDaemon(true)
+            .build());
+    safeModeLogExecutor.scheduleAtFixedRate(() -> {
+      try {
+        logSafeModeStatus();
+      } catch (Throwable t) {
+        LOG.warn("Safe mode periodic logger encountered an error", t);
+      }
+    }, 0L, safeModeLogIntervalMs, TimeUnit.MILLISECONDS);
+    LOG.info("Started periodic Safe Mode logging with interval {} ms", safeModeLogIntervalMs);
+  }
+
+  private void logSafeModeStatus() {
```

Comment:
Can we make this whole method synchronized instead of trying to isolate the specific parts of it that need to be coordinated with other method calls? This method will run quickly and not block other safemode checks. It also makes it easier to reason about. Right now there could be some strange cases where, for example, we read `safeModeStatus` as "in safemode", but it has left safemode by the time we get to the logging section, in which case the rule states won't match the status.